File: get_data/getrankings.mjs
import dotenv from 'dotenv';
import fetch from 'node-fetch';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

dotenv.config();
const token = process.env.API_TOKEN;

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const outputDir = path.join(__dirname, 'output');
const filePath = path.join(outputDir, 'rankings.json');

if (!fs.existsSync(outputDir)) {
    fs.mkdirSync(outputDir, { recursive: true });
}

fetch('https://api.brawlstars.com/v1/rankings/global/players', {
    headers: {
        'Accept': 'application/json',
        'Authorization': `Bearer ${token}`
    }
})
    .then(response => response.json())
    .then(data => {
        fs.writeFile(path.join(filePath), JSON.stringify(data, null, 2), (err) => {
            if (err) {
                console.error('Error writing to file', err);
            } else {
                console.log('JSON data saved successfully!');
            }
        });
    })
    .catch(err => {
        console.error('Error fetching data', err);
    });

File: get_data/get_rankings.py
import brawlstats
import os
import sys
import io
import json
import pandas as pd
from dotenv import load_dotenv

sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8')

output_dir = os.path.join(os.path.dirname(__file__), 'output')
os.makedirs(output_dir, exist_ok=True)
file_path = os.path.join(output_dir, 'rankings.json')

f = open(file_path, 'r', encoding='utf-8')
data = json.load(f)

tags = []
for player in data['items']:
    tags.append(player['tag'])

print(len(tags))
print(tags)